src/js/popup.js

```diff
@@ -1,107 +1,119 @@
+                    );
+                    $('#currentTabTitle').val(
+                        ALL_NOTES_OBJECT[CURRENT_TAB_URL].noteTitle
+                    );
+                    $('#notes-content').val(
+                        ALL_NOTES_OBJECT[CURRENT_TAB_URL].noteText
+                    );
+                }
+            }
+        );
+    };
+
+    const showSuccessAlert = function () {
         $('.alerts').html(
             '<div class="alert alert-success" role="alert">Saved!</div>'
         );
-    }
+    };
 
-    function showErrorAlert() {
+    const showErrorAlert = function () {
         $('.alerts').html(
-            '<div class="alert alert-danger" role="alert">Fail :(</div>'
+            '<div class="alert alert-success" role="alert">Error in Saving!</div>'
         );
-    }
+    };
 
-    function saveNotesForPage() {
-        // https://developer.chrome.com/docs/extensions/reference/storage/
-        if (CURRENT_URL === null) {
-            showErrorAlert();
-        } else if (NOTES_URLS === null) {
+    const saveNotesForCurrentTab = function () {
+        if (CURRENT_TAB_URL === '') {
             showErrorAlert();
         } else {
-            const value = $('#notes-content').val();
-
-            // TODO: encrypt the value
-            // TODO: maybe also encrypt the key (url)
-            const kv = {};
-            kv[CURRENT_URL] = value;
-
-            NOTES_URLS[CURRENT_URL] = true;
-
-            kv[KEY_NOTES_URLS] = NOTES_URLS;
-
-            chrome.storage.sync.set(kv, function () {
-                console.log(CURRENT_URL, value);
-                showSuccessAlert();
-            });
+            const noteUrl = $('#currentTabURL').val();
+            const noteImage = $('#currentTabImage').attr('src');
+            const noteTitle = $('#currentTabTitle').val();
+            const noteText = $('#notes-content').val();
+            // TODO: Date and Time Properties
+
+            const noteMetaDataObject = {
+                noteUrl: noteUrl,
+                noteImage: noteImage,
+                noteTitle: noteTitle,
+                noteText: noteText,
+            };
+
+            ALL_NOTES_OBJECT[CURRENT_TAB_URL] = noteMetaDataObject;
+
+            chrome.storage.sync.set(
+                { CURRENT_TAB_URL: noteMetaDataObject },
+                function () {
+                    showSuccessAlert();
+                }
+            );
+            chrome.storage.sync.set(
+                { CHROME_STORAGE_ALL_NOTES_KEY: ALL_NOTES_OBJECT },
+                function () {
+                    showSuccessAlert();
+                }
+            );
         }
-    }
+    };
 
-    function getPreviouslySavedNotes() {
-        // https://developer.chrome.com/docs/extensions/reference/storage/
+    const fetchAllPreviousNotes = function () {
         chrome.storage.sync.get(
-            [KEY_NOTES_URLS, CURRENT_URL],
+            ['CHROME_STORAGE_ALL_NOTES_KEY'],
```

Comment:
this needs to be the constant, not a string

Reply:
I tried using a constant and it was not fetching properly, but when I use a string for the call it is working properly.

Reply:
Yeah, something isn't working properly, then...